### PR TITLE
test: check nil vm and empty authority

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by DimitrisJim — APPROVED** [review]

File: modules/light-clients/08-wasm/keeper/keeper.go [hunk 1]
```diff
@@ -38,10 +39,18 @@ func NewKeeperWithVM(
...
+		panic("wasm VM must be not nil")
+	}
+
 	if types.WasmVM != nil && !reflect.DeepEqual(types.WasmVM, vm) {
 		panic("global Wasm VM instance should not be set to a different instance")
 	}
 
+	if strings.TrimSpace(authority) == "" {
+		panic("authority must be non-empty")
+	}
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/keeper/keeper.go:52]: should we panic with errors instead? then we test using `suite.RequirePanicsWithError`?

I guess it doesn't really make a difference as a panic is a panic, but seems like it makes sense to panic with an error. Afaik other `NewKeeper` funcs are doing this atm, although its probably not 100% consistent across the board.

**Review by damiannolan — APPROVED** [review]

File: modules/light-clients/08-wasm/keeper/keeper.go (hunk 1 above)
Comment (crodriguezvega) [reply modules/light-clients/08-wasm/keeper/keeper.go:52]: Indeed, I think it's not consistent in the codebase, but I can for now change that here.

File: modules/light-clients/08-wasm/keeper/keeper.go [hunk 2]
```diff
@@ -38,8 +39,16 @@ func NewKeeperWithVM(
 	authority string,
 	vm *wasmvm.VM,
 ) Keeper {
+	if vm == nil {
+		panic(fmt.Errorf("wasm VM must be not nil"))
```
Comment (chatton) [inline modules/light-clients/08-wasm/keeper/keeper.go:43]: [q] should we use `errors.New` if we aren't passing format args?

**Review by chatton — APPROVED** [review]
Changes look good, seems the linter is failing though!

**Pull request merged**